RDKDEV-986 - Upstream FrameRate default values updated changes

tkathi259 opened the pull request:
Reason for change: Updated default values to max and min variable when there is no fps update. As this is generic change, need to upstream the AMLS905X4-275-Fix-FrameRate-onFpsEvent-getting-negative-values.patch file.

Risks: Low

Test Procedure: Sequence of Curl command to check the fps.

curl --header "Content-Type: application/json" --request POST --data '{"jsonrpc": "2.0","id": 1234567890,"method": "Controller.1.activate","params": {"callsign": "org.rdk.FrameRate"}}' http://127.0.0.1:9998/jsonrpc
curl --header "Content-Type: application/json" --request POST --data '{"jsonrpc": "2.0","id": 1234567890,"method": "Controller.1.activate","params": {"callsign": "LightningApp"}}' http://127.0.0.1:9998/jsonrpc
curl --header "Content-Type: application/json" --request POST --data '{"jsonrpc": "2.0","id": 1234567890,"method": "LightningApp.1.url","params": "[https://www.testufo.com/"}](https://www.testufo.com/%22%7D)' http://127.0.0.1:9998/jsonrpc
curl --header "Content-Type: application/json" --request POST --data '{"jsonrpc": "2.0","id": 1234567890,"method": "LightningApp.1.fps"}' http://127.0.0.1:9998/jsonrpc
curl --header "Content-Type: application/json" --request POST --data '{"jsonrpc":"2.0","id":"3","method": "org.rdk.FrameRate.1.setCollectionFrequency", "params": {"frequency": 10000}}' http://127.0.0.1:9998/jsonrpc
curl --header "Content-Type: application/json" --request POST --data '{"jsonrpc":"2.0","id":"3","method": "org.rdk.FrameRate.1.startFpsCollection"}' http://127.0.0.1:9998/jsonrpc

On grepping the OnFPSChanged event in wpeframework.log, able to see fps default value:
grep -irn "onFpsEvent" /opt/logs/wpeframework.log

anand-ky commented on FrameRate/FrameRate.cpp:
```diff
@@ -381,7 +380,7 @@ namespace WPEFramework
                 if (m_numberOfFpsUpdates > 0)
                 {
-                averageFps = (m_totalFpsValues / m_numberOfFpsUpdates);
+                averageFps = (m_totalFpsValues / 2);
```
Checking why this is changed to 2. averageFps is meant to return (m_totalFpsValues / m_numberOfFpsUpdates)

anand-ky commented on FrameRate/FrameRate.cpp:
```diff
@@ -407,7 +406,7 @@ namespace WPEFramework
                 m_minFpsValue = newFpsValue;
             }
-            m_totalFpsValues += newFpsValue;
+	    m_totalFpsValues = m_maxFpsValue + m_minFpsValue;
```
Why is this changed to m_maxFpsValue + m_minFpsValue? The m_totalFpsValues is supposed to track the fps updates.

anand-ky reviewed (changes requested):
This looks like changes being done to fix the case when there are no fps updates. We need to understand the original issue for which this fix was done. If there are no calls to updateFps between calls to startFpsCollection & StopFpsCollection then default values (-1) are expected in the onFPSEvent.

tkathi259 commented:
@anand-ky, we checked your review comments on the Framerate changes. We added these changes to prevent negative values when there is no fps update. It's mainly added to pass the TDK testcase.

Recently we found a similar issue in RDKCOM ticket [RDKCOM-2149](https://ccp.sys.comcast.net/browse/RDKCOM-2149). In that, added the below patch file to update the default values,
https://gerrit.teamccp.com/#/c/608115/

This patch is currently not available in the rdk-next branch, so we added it manually and verified it; however, the default values have not been updated, and we are still getting negative values for min, max, and average when there is no FPS update.

Are you aware of this ticket? Could you please share the conclusion of this issue?

anand-ky commented:
If this was done to pass the TDK testcase, pls change the test case to expect negative values when there is no update.

tkathi259 commented:
According to the component owner's response, negative values are expected when there is no fps update. It will refer to no updates. So we've abandoned the PR changes and will remove the patch. will request to change TDK script based on it.
